crates/ruff/tests/format.rs

```diff
@@ -90,6 +90,172 @@ fn format_warn_stdin_filename_with_files() {
     "###);
 }
 
+#[test]
+fn nonexistent_config_file() {
+    assert_cmd_snapshot!(Command::new(get_cargo_bin(BIN_NAME))
+        .args(["format", "--config", "foo.toml", "."]), @r###"
+        success: false
+        exit_code: 2
+        ----- stdout -----
+
+        ----- stderr -----
+        error: invalid value 'foo.toml' for '--config <CONFIG_OPTION>'
+
+          tip: The `--config` flag must either be a path to a `.toml` configuration file or a TOML `<KEY> = <VALUE>` pair overriding a specific config setting
+          tip: The path `foo.toml` does not exist on your filesystem
+
+        For more information, try '--help'.
+       "###);
+}
+
+#[test]
+fn config_override_rejected_if_invalid_toml() {
+    assert_cmd_snapshot!(Command::new(get_cargo_bin(BIN_NAME))
+        .args(["format", "--config", "foo = bar", "."]), @r###"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    error: invalid value 'foo = bar' for '--config <CONFIG_OPTION>'
+
+      tip: The `--config` flag must either be a path to a `.toml` configuration file or a TOML `<KEY> = <VALUE>` pair overriding a specific config setting
+      tip: The following error occurred when attempting to parse `foo = bar` as a `ruff.toml` config option:
```

Comment:
It feels weird that these are separate tips, is it feasible to combine the message into one?

Similar to https://github.com/astral-sh/ruff/pull/9599/files#r1480584476 you should use "configuration option" consistently.

Reply:
Had a go at combining them into a single tip in https://github.com/astral-sh/ruff/pull/9599/commits/88e7273abf26f1f38e970a058699aa293737a0a4. I also introduced some more line breaks, as I realised the error messages looked terrible if the terminal had a narrow width